src/main.rs

```diff
@@ -348,6 +330,41 @@ fn find_ram_in_linker_script(linker_script: &str) -> Option<MemoryEntry> {
     None
 }
 
+fn arithmetic_op(line: &str) -> u64 {
+    let segments: Vec<&str> = line.split('+').map(|s| s.trim().trim_end()).collect();
+    // if some numbers are hex we convert here
+
+    let mut total_length = 0;
+    for segment in segments {
+        println!("current segment {}", segment);
+        let boundary_pos = segment
+            .find(|c| c == 'K' || c == 'M')
+            .unwrap_or(segment.len());
+        println!("boundary pos {}", boundary_pos);
+        const HEX: &str = "0x";
+        let length: u64 = if segment.starts_with(HEX) {
+            tryc!(u64::from_str_radix(&segment[HEX.len()..boundary_pos], 16))
+        } else {
+            tryc!(segment[..boundary_pos].parse())
+        };
```

Comment:
Since it was like that before you don't have to address this here, but I am just wondering why we are once using `u64::from_str_radix` and the other time `.parse()` to seemingly to the same thing.